src/components/NcActionButton/NcActionButton.vue

```diff
@@ -280,6 +420,66 @@ export default {
 		isFocusable() {
 			return !this.disabled
 		},
+
+		/**
+		 * The current "checked" or "pressed" state for the model behavior
+		 */
+		isChecked() {
+			if (this.modelBehavior === 'radio') {
+				return this.modelValue === this.radioValue
+			}
+			return this.modelValue
+		},
+
+		/**
+		 * HTML attributes to bind to the <button>
+		 */
+		buttonAttributes() {
+			const attributes = {
+				'aria-label': this.ariaLabel,
+				title: this.title,
+				type: 'button',
+			}
+
+			if (this.isInSemanticMenu) {
+				// By default it needs to be a menu item in semantic menus
+				attributes.role = 'menuitem'
+
+				if (this.modelBehavior === 'radio') {
+					attributes.role = 'menuitemradio'
+					attributes['aria-checked'] = this.isChecked ? 'true' : 'false'
```

Comment:
**proposal:** to make it a little bit simple to understand what the value is here.
`isChecked` is defined for all cases (with condition), but actually is used only here. And when it is used only here, it is as simple as one condition.

```suggestion
					attributes['aria-checked'] = this.modelValue === this.radioValue ? 'true' : 'false'
```

Reply:
Yes but there are more places, at least 3 which need the condition for radios

Reply:
My bad, missed it